src/rust/ide/src/controller/graph.rs

```diff
@@ -257,6 +255,29 @@ mod tests {
         }
     }
 
+    #[wasm_bindgen_test]
+    fn node_operations() {
+        TestWithLocalPoolExecutor::set_up().run_test(async {
+            let transport    = MockTransport::new();
+            let file_manager = file_manager_client::Handle::new(transport);
+            let parser       = Parser::new().unwrap();
+            let location     = module::Location("Test".to_string());
+            let code         = "main = Hello World";
+            let idmap        = default();
+            let module       = module::Handle::new_mock
+                (location,code,idmap,file_manager,parser).unwrap();
+            let pos          = module::Position {vector:Vector2::new(0.0,0.0)};
+            let crumbs       = vec![DefinitionName::new_plain("main")];
+            let graph        = graph::Handle::new(module, Id {crumbs}).unwrap();
+
+            let uid          = graph.all_node_infos().unwrap()[0].id();
```

Comment:
What is this `[0]` ? Thsi should be refactored to some variables telling what this 0 is for.

Reply:
Plus, iss there a guarantee that this unwrap and [0] will not fail? If so, what is the guarantee? We should nto have non-total code.